addons/website/models/website_menu.py

```diff
@@ -62,6 +62,28 @@ def name_get(self):
             res.append((menu.id, menu_name))
         return res
 
+    def _validate_parent_menu(self):
+        for record in self:
+            parent_menu = record.parent_id.sudo() if record.parent_id else None
+
+            if parent_menu:
+                # Check hierarchy level
+                if parent_menu.parent_id and parent_menu.parent_id.parent_id:
+                    raise UserError(_("Menus cannot have more than two levels of hierarchy."))
+
+                # Check mega menu conditions
+                if record.is_mega_menu:
+                    if parent_menu.parent_id:
+                        raise UserError(_("A mega menu cannot have a parent menu."))
+                    if record.child_id:
+                        raise UserError(_("A mega menu cannot have a child menu."))
+                elif parent_menu.is_mega_menu:
+                    raise UserError(_("A mega menu cannot have a child menu."))
+
+                # Check for child menu condition
+                if record.child_id and (parent_menu.parent_id or any(child.child_id for child in record.child_id)):
+                    raise UserError(_("Menus with child menus cannot be added as a submenu."))
```

Comment:
Can we simplify it?

Reply:
Is this condition okay?
if record.child_id and (parent_menu.parent_id or record.child_id.child_id):